src/swell/tasks/bufr_to_ioda.py

```diff
@@ -0,0 +1,287 @@
+# (C) Copyright 2021- United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+import os
+import subprocess
+
+from swell.utilities.datetime_util import datetime_formats
+#from swell.utilities.copy_to_dst_dir import copy_to_dst_dir
+from swell.tasks.base.task_base import taskBase
+
+import datetime
+
+import glob
+import yaml
+# --------------------------------------------------------------------------------------------------
+
+# Dictionary linking each obs type to the appropriate yaml template
+#path_to_ioda_conv_yaml_tmpl_dir = os.path.join(self.experiment_path(), 'configuration/iodaconv')
+
+#bufr_to_ioda_yaml_obs_type_dict = {
+bufr2ioda_obs_type_dict = {
+    '1bmhs': 'bufr_ncep_1bmhs.yaml',
+    'amsua': 'bufr_ncep_1bamua_ta.yaml',
+    '1bamua': 'bufr_ncep_1bamua_ta.yaml',
+    'atms': 'bufr_ncep_atms.yaml',
+    'mtiasi': 'bufr_ncep_mtiasi.yaml',
+    'satwind': 'bufr_ncep_satwind_avhrr.yaml',
+    'aircft': 'bufr_ncep_prepbufr_aircft.yaml',
+    'sevcsr': 'bufr_ncep_sevcsr.yaml'
+}
+
+
+# python split filename by delimiter period and then search for string in the resulting list
+def find_obstype_match(filename):
+    """
+    Splits the filename by '.' and returns the first match found in obs_type_search_dict.
+    Prints the match if found, otherwise prints 'No match found.'
+
+    Dictionary:
+    bufr2ioda_obs_type_dict = {
+        '1bmhs': 'bufr_ncep_1bmhs.yaml',
+        'amsua': 'bufr_ncep_1bamua_ta.yaml',
+        '1bamua': 'bufr_ncep_1bamua_ta.yaml',
+        'atms': 'bufr_ncep_atms.yaml',
+        'mtiasi': 'bufr_ncep_mtiasi.yaml',
+        'satwind': 'bufr_ncep_satwind_avhrr.yaml',
+        'aircft': 'bufr_ncep_prepbufr_aircft.yaml',
+        'sevcsr': 'bufr_ncep_sevcsr.yaml'
+    }
+    
+    """
+    # 
+    obs_type_search_dict = bufr2ioda_obs_type_dict
+    parts = filename.split('.')
+    for part in parts:
+        if part in obs_type_search_dict:
+            print(f"Match found: {part}")
+            return part
+    print("No match found.")
+    return None
```

Comment:
Why not use the dict directly?

```suggestion
    parts = filename.split('.')
    for part in parts:
        if part in bufr2ioda_obs_type_dict:
            print(f"Match found: {part}")
            return part
    print("No match found.")
    return None
```

Also try using `self.logger.info` instead of `print` for consistency.